crates/crypto/src/hash/mod.rs

```diff
@@ -39,12 +40,45 @@ impl FromStr for HashAlgorithm {
     }
 }
 
-pub trait SupportedDigest: Digest + private::Sealed {
+static EMPTY_TREE_HASH: Lazy<Vec<Hash<Sha256>>> = Lazy::new(|| {
+    let mut v: Vec<Hash<Sha256>> = Vec::with_capacity(257);
+    fn empty_tree_hash<D: SupportedDigest>(v: &mut Vec<Hash<D>>, height: u32) -> Hash<D> {
+        let hash: Hash<D> = if height == 0 {
+            hash_empty()
+        } else {
+            let last_hash = empty_tree_hash(v, height - 1);
+            hash_branch(&last_hash, &last_hash)
+        };
+        v.push(hash.clone());
+        hash
+    }
+    empty_tree_hash(&mut v, 256);
+    v
+});
+
+// If updating this function, also update `hash_empty` in transparency map
+pub(crate) fn hash_empty<D: SupportedDigest>() -> Hash<D> {
```

Comment:
Copy `hash_leaf` over here and update this `hash_empty` to call it like the suggestion in the other location.